src/Avalonia.Visuals/Media/HsvColor.cs

```diff
@@ -0,0 +1,493 @@
+                    r = max;
+                    g = min + (chroma * intermediateColorPercentage);
+                    b = min;
+                    break;
+                case 1:
+                    r = min + (chroma * (1 - intermediateColorPercentage));
+                    g = max;
+                    b = min;
+                    break;
+                case 2:
+                    r = min;
+                    g = max;
+                    b = min + (chroma * intermediateColorPercentage);
+                    break;
+                case 3:
+                    r = min;
+                    g = min + (chroma * (1 - intermediateColorPercentage));
+                    b = max;
+                    break;
+                case 4:
+                    r = min + (chroma * intermediateColorPercentage);
+                    g = min;
+                    b = max;
+                    break;
+                case 5:
+                    r = max;
+                    g = min;
+                    b = min + (chroma * (1 - intermediateColorPercentage));
+                    break;
+            }
+
+            return Color.FromArgb(
+                (byte)Math.Round(alpha * 255),
+                (byte)Math.Round(r * 255),
+                (byte)Math.Round(g * 255),
+                (byte)Math.Round(b * 255));
+        }
+
+        /// <summary>
+        /// Converts the given RGB color to it's HSV color equivalent.
+        /// </summary>
+        /// <param name="color">The color in the RGB color model.</param>
+        /// <returns>A new <see cref="HsvColor"/> equivalent to the given RGBA values.</returns>
+        public static HsvColor FromRgb(Color color)
+        {
+            return HsvColor.FromRgb(color.R, color.G, color.B, color.A);
+        }
+
+        /// <summary>
+        /// Converts the given RGBA color channel values to it's HSV color equivalent.
+        /// </summary>
+        /// <param name="red">The red channel value in the RGB color model.</param>
+        /// <param name="green">The green channel value in the RGB color model.</param>
+        /// <param name="blue">The blue channel value in the RGB color model.</param>
+        /// <param name="alpha">The alpha channel value.</param>
+        /// <returns>A new <see cref="HsvColor"/> equivalent to the given RGBA values.</returns>
+        public static HsvColor FromRgb(
+            byte red,
+            byte green,
+            byte blue,
+            byte alpha = 0xFF)
+        {
+            // Note: Conversion code is originally based on the C++ in WinUI (licensed MIT)
+            // https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/Common/ColorConversion.cpp
+            // This was used because it is the best documented and likely most optimized for performance
+            // Alpha channel support was added
+
+            // Normalize RGBA channel values into the 0..1 range used by this algorithm
+            double r = red / 255.0;
```

Comment:
perhaps `255d` instead of `255.0`? Idk, optional :)

Reply:
Just developer preference really. 255.0 is more readable IMO